File: 프로그래머스/3/12938. 최고의 집합/최고의 집합.js
function solution(n, s) {
    
    var div = Math.floor(s/n);
    var mod = s%n;
    
    if(div == 0) return [-1];
    
    var answer = Array(n).fill(0);
    
    for(let i=n-1; i>=0; i--){
        if(mod > 0){
            answer[i] = div+1;
            mod--;
        }else{
            answer[i] = div;
        }
    }
    
    return answer;
}

File: 프로그래머스/3/12938. 최고의 집합/최고의 집합.java
class Solution {
    public int[] solution(int n, int s) {
        
        int div = s/n;
        int left = s%n;
        
        if(div == 0) return new int[] {-1};
        
        int answer[] = new int[n];
        
        for(int i=n-1; i>=0; i--){
            if(left > 0){
                answer[i] = div+1;
                left--;
            }else{
                answer[i] = div;
            }
        }

        return answer;
    }
}